includes/class-sensei-course-structure.php

```diff
@@ -854,4 +862,39 @@ public function get_first_incomplete_lesson_id() {
 
 		return false;
 	}
+
+	/**
+	 * Get module for teacher by slug.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param string $slug             The probable slug.
+	 * @param array  $additional_slugs Additional slugs to search with.
+	 *
+	 * @return WP_Term|null The found term or null.
+	 */
+	private function get_module_for_teacher_by_slug( string $slug, array $additional_slugs = [] ) {
+		$search_slugs  = [] + $additional_slugs ?? [];
+		$user_id       = get_post( $this->course_id )->post_author;
+		$is_admin_user = user_can( $user_id, 'manage_options' );
+		$split_slug    = explode( '-', $slug );
+
+		array_shift( $split_slug );
+
+		if ( ! $is_admin_user ) {
+			$search_slugs[] = $user_id . '-' . $slug;
+			$search_slugs[] = $user_id . '-' . implode( '-', $split_slug );
+		} elseif ( count( $split_slug ) > 0 ) {
+			$search_slugs[] = implode( '-', $split_slug );
+		}
+
+		foreach ( $search_slugs as $search_slug ) {
+			$search_term = get_term_by( 'slug', $search_slug, 'module' );
+
+			if ( $search_term && ! is_wp_error( $search_term ) ) {
+				return $search_term;
+			}
+		}
+		return null;
+	}
```

Comment:
Sorry, but I'm a bit confused by this method. Could you add a few code comments explaining what the logic is?